lisa/features/gpu.py

```diff
@@ -0,0 +1,148 @@
+
+    # download and install CUDA Driver
+    def _install_cuda_driver(self, version: str = "10.1.105-1") -> None:
+        self._log.debug("Starting CUDA driver installation")
+        cuda_repo = ""
+
+        os_version = self._node.os.os_version
+
+        # CUDA driver installation for redhat distros
+        if isinstance(self._node.os, Redhat):
+            release = os_version.release.split(".")[0]
+            cuda_repo_pkg = f"cuda-repo-rhel{release}-{version}.x86_64.rpm"
+            cuda_repo = (
+                "http://developer.download.nvidia.com/"
+                f"compute/cuda/repos/rhel{release}/x86_64/{cuda_repo_pkg}"
+            )
+        # CUDA driver installation for Ubuntu distros
+        elif isinstance(self._node.os, Ubuntu):
+            release = re.sub("[^0-9]+", "", os_version.release)
+            cuda_repo_pkg = f"cuda-repo-ubuntu{release}_{version}_amd64.deb"
+            cuda_repo = (
+                "http://developer.download.nvidia.com/compute/"
+                f"cuda/repos/ubuntu{release}/x86_64/{cuda_repo_pkg}"
+            )
+        else:
+            raise LisaException(
+                f"Distro {self._node.os.__class__.__name__}"
+                "not supported to install CUDA driver."
+            )
+
+        # download and install the cuda driver package from the repo
+        self._node.os._install_package_from_url(f"{cuda_repo}", signed=False)
+
+    def _install_gpu_dep(self) -> None:
+        uname_tool = self._node.tools[Uname]
+        uname_ver = uname_tool.get_linux_information().uname_version
+
+        # install dependency libraries for redhat and CentOS
+        if isinstance(self._node.os, Redhat):
+            # install the kernel-devel and kernel-header packages
+            package_name = f"kernel-devel-{uname_ver} kernel-headers-{uname_ver}"
+            self._node.os.install_packages(package_name)
+            # mesa-libEGL install/update is require to avoid a conflict between
+            # libraries - bugzilla.redhat 1584740
+            package_name = "mesa-libGL mesa-libEGL libglvnd-devel"
+            self._node.os.install_packages(package_name)
+            # install dkms
+            package_name = "dkms"
+            self._node.os.install_packages(package_name, signed=False)
+        # install dependency libraries for Ubuntu
+        elif isinstance(self._node.os, Ubuntu):
+            package_name = (
+                f"build-essential libelf-dev linux-tools-{uname_ver}"
+                f" linux-cloud-tools-{uname_ver} python libglvnd-dev ubuntu-desktop"
+            )
+            self._node.os.install_packages(package_name)
+        else:
+            raise LisaException(
+                f"Distro {self._node.os.__class__.__name__}"
+                " is not supported for GPU."
+            )
+
+    def check_support(self) -> None:
+        # TODO: more supportability can be defined here
+        if not self._is_supported:
```

Comment:
It's defined as a method, but here it's used like a property. It will returns true always. So either change it to property, or call it here like a method.

Reply:
Yeah missed it after changing the property to a method. Thanks for noting.